core/src/main/java/org/kiwix/kiwixmobile/core/main/MainPresenter.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.util.Log
+import io.reactivex.CompletableObserver
+import io.reactivex.SingleObserver
+import io.reactivex.disposables.Disposable
+import org.kiwix.kiwixmobile.core.base.BasePresenter
+import org.kiwix.kiwixmobile.core.bookmark.BookmarkItem
+import org.kiwix.kiwixmobile.core.data.DataSource
+import org.kiwix.kiwixmobile.core.di.ActivityScope
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import org.kiwix.kiwixmobile.core.main.MainContract.Presenter
+import org.kiwix.kiwixmobile.core.main.MainContract.View
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskListItem
+import org.kiwix.kiwixmobile.core.zim_manager.fileselect_view.adapter.BooksOnDiskListItem.BookOnDisk
+import javax.inject.Inject
+
+@ActivityScope
+internal class MainPresenter @Inject constructor(private val dataSource: DataSource) :
+  BasePresenter<View?>(), Presenter {
+  override fun loadBooks() {
+    dataSource.languageCategorizedBooks
```

Comment:
So we can rewrite this a little bit to be more similar to standard usage of Rx.

it should look like

```
compositeDisosable.add(
   dataSource.languageCategorizedBooks
      .subscribe(
        { view!!.addBooks(it) },
        { Log.e(TAG, "unable to load books", it) }
      )
)
```

And that will make it much conciser and easier to read